src/iqm/benchmarks/benchmark_experiment.py

```diff
@@ -52,15 +53,14 @@ def __init__(
         """
         self.timestamp = strftime("%Y%m%d-%H%M%S")
 
-        self.device_id = device_id if device_id is not None else backend
-
         if isinstance(backend, str):
             self.backend = get_iqm_backend(backend)
-
         else:
             assert isinstance(backend, IQMBackendBase)
             self.backend = backend
 
+        self.device_id = device_id if device_id is not None else self.backend.name
+
```

Comment:
Is this what fixes the output folder name error?

Reply:
Yes indeed, before it was trying to assign `backend` when it was passed as a backend object (and below the folder name has `device_id` in it)